agent/stats/utils_unix.go

```diff
@@ -43,7 +43,11 @@ func dockerStatsToContainerStats(dockerStats *types.StatsJSON) (*ContainerStats,
 	}, nil
 }
 
+// Safe return of storage stats based on Docker Stats (Unix)
 func getStorageStats(dockerStats *types.StatsJSON) (uint64, uint64) {
+	if dockerStats.BlkioStats.IoServiceBytesRecursive == nil {
+		seelog.Debug("storage stats (Unix) not reported for container")
+	}
```

Comment:
should you return here? if there's no storage stats to report, why are we proceeding to the storage stats logic below?

Reply:
The point of this change is to ensure that we have uint64(0) as our null value rather than breaking for read and write block stats.  The rest of the function exists to initialize the return values, then update them (unless they aren't available -- in which case we return 0).
This is in line with what we are doing for network stats.  See the `getNetworkStats() ` in `agent/stats/utils.go`.

Reply:
Updated based on https://github.com/moby/moby/blob/master/api/types/stats.go#L111-L114 and the zero-initialization of the StorageStats struct.  We are guaranteed zero uint64 values if the stats object exists.  The windows StorageStats struct is initialized here:
https://github.com/moby/moby/blob/master/daemon/daemon_windows.go#L579-L582